src/adv.py

```diff
@@ -56,18 +56,45 @@
 
 # Write a class to hold player information, e.g. what room they are in currently
 
+class Player:
+    def __init__(self, name, starting_room):
+        self.name = name
+        self.room = starting_room
+
 #
 # Main
 #
 
 # Make a new player object that is currently in the 'outside' room.
 
+player1 = Player('player1', rooms["outside"])
+
 # Write a loop that:
 #
 # * Prints the current room name
 # * Prints the current description (the textwrap module might be useful here).
 # * Waits for user input and decides what to do.
 #
+
+done = False
+while not done:
+    print(player1.room["name"])
+    print(player1.room["description"])
+
+    usr_in = input("What would you like to do?")
+
+    if usr_in == "q":
+        done = True
+    elif usr_in in ["n", "s", "w", "e"]:
+        try:
+            new_room = player1.room[usr_in + "_to"]
```

Comment:
This is great! Sad to say I'm looking for tips to my own implementation here and I really like this.